### PR TITLE
BUG: Allow tag commit hashes for graphs

### PR DESCRIPTION
**Pull request by HaoZeke**
Closes https://github.com/airspeed-velocity/asv/issues/1226.

cc @mattip

### CODE REVIEW

**Comment by mattip** [comment]
There is a test failure on appveyor. Thoughts?

**Comment by mattip** [comment]
On PR #1221 the github CI also failed, but appveyor passed.

**Comment by HaoZeke** [comment]
> There is a test failure on appveyor. Thoughts?

Sorry, yes I'll need to update the tests.

**Comment by mattip** [comment]
There are some [linting failures](https://github.com/airspeed-velocity/asv/actions/runs/3812516562/jobs/6492427747#step:4:59) after #1229

**Comment by mattip** [comment]
... and a failing test

**Comment by mattip** [comment]
Still failing CI ...

**Comment by HaoZeke** [comment]
Sorry, had to setup #1232 before I could debug this properly, will rebase with a fix in a bit.

File: test/tools.py [hunk 1]
```diff
@@ -210,7 +210,11 @@ def get_hash(self, name):
 
     def get_branch_hashes(self, branch=None):
         if branch is None:
-            branch = "master"
+            try:
+                branch = self.run_git(["config", "init.defaultBranch"],
+                                      display_error=False).strip()
+            except util.ProcessError:
+                branch = 'master'
```
Comment (mattip) [inline test/tools.py:217]: At some point it might be nice to refactor this into a helper function

**Comment by mattip** [comment]
Thanks @HaoZeke

**Pull request merged**